grails-app/services/org/pih/warehouse/core/MailService.groovy

```diff
@@ -53,353 +53,157 @@ class MailService {
+    Boolean sendHtmlMailWithAttachment(User fromUser, Collection toList, Collection ccList, String subject, String body, byte[] bytes, String name, String mimeType, Integer port) {
+        return sendMailImpl(subject, body, fromUser?.email, fromUser?.name, toList, ccList, Collections.singleton(new Attachment(name: name, mimeType: mimeType, bytes: bytes)), port, true, false)
     }
 
-    /**
-     *
-     * @param fromUser
-     * @param toList
-     * @param ccList
-     * @param subject
-     * @param body
-     * @param bytes
-     * @param name
-     * @param mimeType
-     * @params port* @return
-     */
-    def sendHtmlMailWithAttachment(User fromUser, Collection toList, Collection ccList, String subject, String body, byte[] bytes, String name, String mimeType, Integer port) {
-        List<Attachment> attachments = []
-        Attachment attachment = new Attachment(name: name, mimeType: mimeType, bytes: bytes)
-        attachments.add(attachment)
-        sendHtmlMailWithAttachment(fromUser, toList, ccList, subject, body, attachments, port)
+    Boolean sendHtmlMailWithAttachment(User fromUser, Collection toList, Collection ccList, String subject, String body, List<Attachment> attachments, Integer port) {
+        return sendMailImpl(subject, body, fromUser?.email, fromUser?.name, toList, ccList, attachments, port, true, false)
     }
 
     /**
+     * Send an email.
      *
-     * @param toList
-     * @param ccList
      * @param subject
      * @param body
+     * @param from
+     * @param fromName
+     * @param to
+     * @param cc
      * @param attachments
      * @param port
-     * @return
+     * @param useHtml
+     * @param override
+     * @return true if an email was sent, false otherwise.
      */
-    def sendHtmlMailWithAttachment(User fromUser, Collection toList, Collection ccList, String subject, String body, List<Attachment> attachments, Integer port) {
-        log.info("Sending email with attachment " + toList)
-        if (isMailEnabled) {
-            try {
-                // Create the email message
-                HtmlEmail email = new HtmlEmail()
-                email.setCharset("UTF-8")
-                email.setHostName(defaultHost)
-
-                // Override smtp port
-                email.setSmtpPort(port ?: defaultPort)
-
-                email.setFrom(defaultFrom)
-                toList.each { to -> email.addTo(to) }
-                if (ccList) {
-                    ccList.each { cc -> email.addCc(cc) }
-                }
-
-                email.setSubject("${prefix} " + subject)
-                email.setHtmlMsg(body)
-
-                // Override from user
-                if (fromUser) {
-                    email.setFrom(fromUser.email, fromUser.name)
-                }
-
-                // Authenticate
-                if (username && password) {
-                    email.setAuthentication(username, password)
-                }
+    Boolean sendMailImpl(
```

Comment:
Minor nitpick but could we change this to. 
```
Boolean doSendMail()
```
I like that convention better than Impl since Impl is usually used for the a class implementing an interface, like
```
BasicMailServiceImpl implements MailService
```

Reply:
hahaha not a nitpick at all! I originally called this method `actuallySendTheActualMail()` and tried to rename it to be more professional. `doSendMail()` is great. Fixed in ff47dc359.